Core/Object Arts/Dolphin/IDE/Base/MethodWorkspace.cls

```diff
@@ -357,6 +389,41 @@ selectedNode
 	range isEmpty ifTrue: [range := range start to: range start].
 	^(self nodeForRange: range) ifNil: [self parseTree]!
 
+selectionChanged
+	"The receiver highlights all text matching <view selection>"
+
+	| index indicators selection selectionStart style tag text |
+	tag := 'secondary selection'.
+	style := self secondarySelectionStyle.
+	indicators := view indicators.
+	indicators copy do: [:each | 
+		each tag = tag ifTrue: [
+			indicators remove: each.
+		].
+	].
+	((selection := view selection) isNil or: [selection isEmpty]) ifTrue: [
+		view indicators: indicators.
+		^self
+	].
+	index := 0.
+	text := view plainText.
```

Comment:
As to performance, would it be reasonable to use a criteria like size of code (say, 10K or 100K characters)? We could do some measurements to determine a reasonable cut-off.

As to the search (and the general implementation), how reasonable would it be to move most of this into ScintillaView so it could be reused by other clients of ScintillaView (such as Jade)? That is, should ScintillaView be restricted to being a pure wrapper around the external DLL or my we add Dolphin-specific capabilities?

Reply:
I think addressing the perf is fine as a later increment (if it proves necessary). 
I agree that adding it to ScintillaView would makes sense, as its generally useful and is View level functionality. It could be done in a layered fashion, e.g. have a method to mark all matches of some search string with an indicator. This should (ideally) use the Scintilla search for flexibility and perf, but that could be a later increment. This would be useful, for example, hooked into the to the find/replace behaviour to show all the potential matches. The method browser could use it to highlight all references to things, etc. Of course you'd also use it to implement the selection matching behaviour. The selection matching ought to be optional and probably off by default if implemented at the Scintilla level.

Reply:
Another thought - if you provide the basic matching functionality, you can then decide whether it is worth putting in the event response. One issue with doing so is that it implies a standard indicator style is present. That's OK, but it may imply that it ought to sit outside. Difficult to be sure without trying it.